CMakeLists.txt

```diff
@@ -24,6 +24,9 @@ else()
     set(VERSION_FULL "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}")
 endif()
 
+# silence boost warnings: "#pragma message: This header is deprecated" from program_options
+add_definitions("-DBOOST_ALLOW_DEPRECATED_HEADERS")
+add_definitions("-DBOOST_BIND_GLOBAL_PLACEHOLDERS")
```

Comment:
For unrelated or broad (e.g. global, such as this) clean up, I'd recommend a separate PR so that if the change is contentious or starts a discussion it doesn't block the unrelated bug/feature changes.

I'd advise against this change for a couple reasons:
1) Specific to this change, the options are enabling and silencing deprecated behavior.
2) More generally, this is the kind of thing that gets left in CMakeLists.txt _forever_ as cruft without any sort of criteria for when it can or should be removed.

I don't see deprecated header warnings from program_options in boost 1.81. And the only boost bind warning I had was trivially fixable (#739). So there does appear to be a point where these options are no longer required (and undesirable to have enabled).

If the internal boost warning is bothersome, maybe moving boost includes to `SYSTEM`/`-isystem` would be an option to silence those warnings? (not terribly a fan of this either, since nominally boost should not be generating warnings itself)

Reply:
I will remove this. 

But I still think that this should be added. Since the recommended build instruction for leap do not specify to use boost 1.81, I think it is likely that people may use other versions of boost, and in my opinion having thousands of warnings when building doesn't reflect well on us.